warehouse/packaging/__init__.py

```diff
@@ -38,13 +40,36 @@ def includeme(config):
     # Register our origin cache keys
     config.register_origin_cache_keys(
         Project,
-        cache_keys=["project/{obj.normalized_name}"],
-        purge_keys=["project/{obj.normalized_name}", "all-projects"],
+        cache_keys=[
+            key_factory("project/{obj.normalized_name}"),
+            key_factory("user/{itr.username}", iterate_on='users'),
```

Comment:
What changes to a user would get reflected on pages that use the ``Project`` model?

Reply:
Technically if the user changes their primary email, their avatar would change, which is shown on the "Maintainers" section for project/release pages.

Also if we ever ship #1190, the URLs to maintainers would be wrong.

Reply:
My primary concern is that there is a limit to how many surrogate keys (what these are) we can attach to a single HTTP response. Since we have no control over how many users can get added to a single project, that would mean that a page may have more surrogate keys than we can support in a single response (and I believe what happens then is that we just silently lose some of the surrogate keys).

I *think* probably it doesn't make sense to support ``iterate_on`` for cache keys because we can't control how many keys are going to get added. However ``iterate_on`` for purge keys makes total sense, and this example here could easily be implemented by, instead of adding a cache key for user to each page, adding a purge key to ``User`` that is something like ``key_factory("project/{itr.normalized_name}", iterate_on="projects")``. That should have the same effect, without the unbounded surrogate key problem.

Reply:
Oh, and if you make that change, add a comment please about why one supports it and one doesn't, otherwise it'll probably be an attractive nuisance for people wanting to add support for it to cache keys and relying on tribal knowledge for why.